src/drivers/console.c

```diff
@@ -18,11 +18,15 @@ struct console_driver *console_drivers[4] = {
     &serial_console_driver,
 };
 
+static u64 write_lock;
+
 void console_write(char *s, bytes count)
 {
+    spin_lock(&write_lock);
```

Comment:
Yes, absolutely. We have the logging with timestamps (log_printf), but it's not used in very early init due to the clocksource not being available. But that would be easy to iron out, and really it ought to have some kind of mux behind it so that we can direct logs to storage or net. I'll file an issue if there isn't one already.